dropwizard-unix-socket/src/test/java/io/dropwizard/unixsocket/UnixSocketConnectorFactoryTest.java

```diff
@@ -0,0 +1,124 @@
+import java.net.UnixDomainSocketAddress;
+import java.nio.channels.Channels;
+import java.nio.channels.SocketChannel;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+@DisabledOnOs(OS.WINDOWS) // Windows has Unix Socket support but not on all versions.
+@ExtendWith(DropwizardExtensionsSupport.class)
+class UnixSocketConnectorFactoryTest {
+    private static final String httpRequest = "GET /app/hello HTTP/1.1\r\n" +
+        "Host: dropwizard-unixsock\r\n" +
+        "\r\n";
+
+    public DropwizardAppExtension<Configuration> ext = new DropwizardAppExtension<>(HelloWorldApplication.class,
+        ResourceHelpers.resourceFilePath("yaml/usock-server.yml"));
+
+    private UnixDomainSocketAddress socketAddress;
+
+    @BeforeEach
+    public void setUp() {
+        final File socket = new File("/tmp/dropwizard.sock");
+        socket.deleteOnExit();
+
+        socketAddress = UnixDomainSocketAddress.of(socket.getPath());
+    }
+
+    @Test
+    public void testClient() throws Exception {
+        try (SocketChannel channel = SocketChannel.open(StandardProtocolFamily.UNIX)) {
+            channel.connect(socketAddress);
+            OutputStream os = Channels.newOutputStream(channel);
+            OutputStreamWriter osw = new OutputStreamWriter(os, UTF_8);
+            PrintWriter writer = new PrintWriter(osw);
+            InputStream is = Channels.newInputStream(channel);
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(is, UTF_8))) {
+                writer.print(httpRequest);
+                writer.flush();
+
+                verifyHelloWorldReceived(reader);
+            }
+        }
+    }
+
+    @Test
+    public void testManyCalls() throws Exception {
+        try (SocketChannel channel = SocketChannel.open(StandardProtocolFamily.UNIX)) {
+            channel.connect(socketAddress);
+            OutputStream os = Channels.newOutputStream(channel);
+            OutputStreamWriter osw = new OutputStreamWriter(os, UTF_8);
+            PrintWriter writer = new PrintWriter(osw);
+            InputStream is = Channels.newInputStream(channel);
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(is, UTF_8))) {
+                for (int i = 0; i < 1000; i++) {
+                    writer.print(httpRequest);
+                    writer.flush();
+
+                    verifyHelloWorldReceived(reader);
+                }
+            }
+        }
+    }
+
+    private static void verifyHelloWorldReceived(BufferedReader reader) throws IOException {
+        for (int i = 0; i < 5; i++) {
+            String line = reader.readLine();
+            if (line.isEmpty()) {
+                // here comes body. Need to be specific otherwise it hangs for a long time.
+                char[] buffer = new char[18];
+                int read = reader.read(buffer, 0, 18);
```

Comment:
Here you'll always only read 18 bytes. If the response is longer, this wouldn't be detected. I'd like to have a check, if the size of the response is equal to 18 bytes or if some bytes will get swallowed.